### PR TITLE
CORS-3483: azure: Update CAPZ version to 1.15.1 

### PR DESCRIPTION
**Pull request by rna-afk**
Updating the CAPZ version to the latest 1.15.1.

### CODE REVIEW

**Comment by sadasu** [comment]
/approve cancel

It appears more work needs to be done https://github.com/openshift/installer/pull/8488#issuecomment-2137767434

File: cluster-api/providers/azure/go.mod [hunk 1]
```diff
@@ -104,52 +97,49 @@ require (
...
+	sigs.k8s.io/cluster-api v1.7.2 // indirect
+	sigs.k8s.io/controller-runtime v0.17.3 // indirect
 	sigs.k8s.io/json v0.0.0-20221116044647-bc3834ca7abd // indirect
-	sigs.k8s.io/structured-merge-diff/v4 v4.3.0 // indirect
+	sigs.k8s.io/structured-merge-diff/v4 v4.4.1 // indirect
 	sigs.k8s.io/yaml v1.4.0 // indirect
 )
 
-replace sigs.k8s.io/cluster-api => sigs.k8s.io/cluster-api v1.6.3
+// replace sigs.k8s.io/cluster-api => sigs.k8s.io/cluster-api v1.6.3
```
Comment (r4f4) [inline cluster-api/providers/azure/go.mod:142]: This is a bad sign. We want the capi versions to match or we might stumble into a hard to find bug.

**Comment by sadasu** [comment]
@rna-afk could you also split this into 2 commits. One for the go.mod changes and another for all the vendoring updates.

File: cluster-api/providers/azure/go.mod (hunk 1 above)
Comment (patrickdillon) [reply cluster-api/providers/azure/go.mod:142]: This version v1.6.3 is leftover installer code. This replace should match the upstream go.mod which is on a more manageable v1.7.2:

https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/901863904deee498347e4e617bc53211e4079f73/go.mod#L219-L222

Comment (r4f4) [reply cluster-api/providers/azure/go.mod:142]: @patrickdillon it worries me that we're building against 1.7.2 but running capi 1.7.0: https://github.com/openshift/installer/blob/master/cluster-api/cluster-api/go.mod#L5. Might not be an issue... until it is. I'd rather we also bumped the installer capi binary to 1.7.2

File: cluster-api/cluster-api/go.mod [hunk 2]
```diff
@@ -2,7 +2,7 @@ module openshift/installer/cluster-api/providers/core
 
 go 1.21
 
-require sigs.k8s.io/cluster-api v1.7.0
+require sigs.k8s.io/cluster-api v1.7.2
```
Comment (r4f4) [inline cluster-api/cluster-api/go.mod:5]: Nice. Let's just make sure we run tests for all capi providers to check this doesn't cause issues.

**Review by r4f4 — COMMENTED** [review]
AFAICT the providers have not been impacted by the capi bump to 1.7.2. I just left one more comment.

**Review by r4f4 — COMMENTED** [review]
/lgtm

**Comment by patrickdillon** [comment]
/approve

We also need:

> to include a commit that updates DisableExtensionOperations on the machines, so that we can test & leverage the updated functionality we will get by vendoring.

in order to close CORS-3483. I'm fine with doing that in a separate PR.

**Comment by rna-afk** [comment]
Created #8594 to set the `DisableExtensionOperations`

**Comment by sadasu** [comment]
/close

Replaced by https://github.com/openshift/installer/pull/8627

**Pull request closed without merging**